### PR TITLE
OBPIH-4358 Display all product associations on Edit product page

### PR DESCRIPTION
**Pull request by DaRoTP**

### CODE REVIEW

File: grails-app/controllers/org/pih/warehouse/product/ProductController.groovy [hunk 1]
```diff
@@ -674,6 +676,26 @@ class ProductController {
...
 
+    def showAssociatedProducts = { StockCardCommand cmd ->
+        def product = Product.get(params.id)
+        def location = Location.get(session?.warehouse?.id)
+
+        def associations = product?.associations
+        def products = product?.associatedProducts() as List
+        def quantityAvailableMap = [:]
+        if (!products.isEmpty()) {
+            quantityAvailableMap = productAvailabilityService.getQuantityAvailableToPromiseByProduct(location, products)
```
Comment (awalkowiak) [inline grails-app/controllers/org/pih/warehouse/product/ProductController.groovy:687]: Why you added this? Was it specifically requested to also have a quantity available to promise?

File: grails-app/controllers/org/pih/warehouse/product/ProductController.groovy [hunk 2]
```diff
@@ -674,6 +676,26 @@ class ProductController {
         }
     }
 
+    def showAssociatedProducts = { StockCardCommand cmd ->
```
Comment (awalkowiak) [inline grails-app/controllers/org/pih/warehouse/product/ProductController.groovy:679]: Why you did not use the `renderTemplate` action from `productController`?

**Review by awalkowiak — CHANGES_REQUESTED** [review]

File: grails-app/controllers/org/pih/warehouse/product/ProductController.groovy (hunk 2 above)
Comment (DaRoTP) [reply grails-app/controllers/org/pih/warehouse/product/ProductController.groovy]: it seems that initially I have misunderstood the task and added similar view for **_showAssociatedProducts** in edit page as on stock card. 
Instead I only had to display all associated products on **_productSubstitutions** page and change some labels

File: grails-app/views/product/_productAssociations.gsp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                            </g:else>
+                        </td>
+
+                    </tr>
+                </g:each>
+                <g:unless test="${productInstance?.associations}">
+                    <tr class="prop">
+                        <td colspan="11">
+                            <div class="padded fade center">
+                                <!--<g:message code="default.empty.message" default="There are no {0}" args="[g.message(code:'product.substitutions.label')]"/>-->
```
Comment (awalkowiak) [inline grails-app/views/product/_productAssociations.gsp:77]: Uncomment and change wording from `substitutions` to `associations`

**Review by awalkowiak — CHANGES_REQUESTED** [review]

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**